aten/src/ATen/native/Convolution.cpp

```diff
@@ -684,7 +684,13 @@ at::Tensor _convolution(
             input.contiguous(), weight, bias,
             padding, stride, dilation, params.groups, params.benchmark, params.deterministic);
       } else {
-          output = at::thnn_conv_depthwise2d(input.contiguous(), weight, kernel_size, bias, stride, padding, dilation);
+          if (input.ndimension() == 4) {
```

Comment:
Do you actually ever get here if cudnn is available?
Should the cudnn detection be modified to not use it in this particular setting?

Reply:
Actually only full precision will go to here, is that okie if add ```input.ndimension() == 4``` condition to ```use_cudnn_depthwise```  function to ensure every 5D tensor input will go to ```conv_depthwise3d```?